src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -1122,11 +1101,18 @@ internal void FlushContentToDisk()
                     {
                         try
                         {
+                            Encoding currentEncoding = null;
+
+                            using (StreamReader reader = new StreamReader(this.Path, Utils.utf8NoBom, true))
+                            {
+                                currentEncoding = reader.CurrentEncoding;
+                            }
+
                             // Try to first open the file with permissions that will allow us to read from it
                             // later.
                             _contentWriter = new StreamWriter(
                                 new FileStream(this.Path, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read),
-                                this.Encoding);
+                                currentEncoding ?? Utils.utf8NoBom);
```

Comment:
Please initialize `currentEncoding`  in line 1104 and remove this `??`.

Reply:
Yes, much better, done